packages/shorebird_cli/lib/src/commands/preview_command.dart

```diff
@@ -0,0 +1,186 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/adb.dart';
+import 'package:shorebird_cli/src/bundletool.dart';
+import 'package:shorebird_cli/src/cache.dart';
+import 'package:shorebird_cli/src/code_push_client_wrapper.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/shorebird_config_mixin.dart';
+import 'package:shorebird_cli/src/shorebird_validation_mixin.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+
+/// {@template preview_command}
+/// `shorebird preview` command.
+/// {@endtemplate}
+class PreviewCommand extends ShorebirdCommand
+    with ShorebirdConfigMixin, ShorebirdValidationMixin {
+  /// {@macro preview_command}
+  PreviewCommand({super.validators}) {
+    argParser
+      ..addOption(
+        'device-id',
+        help: 'The ID of the device or simulator to preview the release on.',
+      )
+      ..addOption(
+        'app-id',
+        help: 'The ID of the app to preview the release for.',
+      )
+      ..addOption(
+        'release-version',
+        help: 'The version of the release (e.g. "1.0.0").',
+      );
+  }
+
+  @override
+  String get name => 'preview';
+
+  @override
+  String get description => 'Preview a specific release on a device.';
+
+  @override
+  Future<int> run() async {
+    try {
+      await validatePreconditions(checkUserIsAuthenticated: true);
+    } on PreconditionFailedException catch (error) {
+      return error.exitCode.code;
+    }
+
+    Future<String?> promptForApp() async {
+      final apps = await codePushClientWrapper.getApps();
+      if (apps.isEmpty) return null;
+      final app = logger.chooseOne(
+        'Which app would you like to preview?',
+        choices: apps,
+        display: (app) => app.displayName,
+      );
+      return app.appId;
+    }
+
+    Future<String?> promptForReleaseVersion(String appId) async {
+      final releases = await codePushClientWrapper.getReleases(appId: appId);
+      if (releases.isEmpty) return null;
+      final release = logger.chooseOne(
+        'Which release would you like to preview?',
+        choices: releases,
+        display: (release) => release.version,
+      );
+      return release.version;
+    }
+
+    const platform = 'android';
```

Comment:
Not something to tackle in this PR, but we should probably make `platform` an enum instead of a string throughout

Reply:
yup agreed that needs to be done throughout